phoenix-core/src/it/java/org/apache/phoenix/mapreduce/VerifyReplicationToolIT.java

```diff
@@ -0,0 +1,323 @@
+        upsertSelectData(sourceTableName, sourceAndTargetTenant, -1, 3);
+        verify(String.format("TENANT_ID = '%s'", sourceAndTargetTenant), 7, 3, 0, 0, 3);
+    }
+
+    @Test
+    public void testVerifyRowsCountMismatch() throws Exception {
+        Connection conn = DriverManager.getConnection(getUrl());
+        // delete one common row from the target
+        PreparedStatement deleteTargetStmt =
+            conn.prepareStatement(String.format("DELETE FROM %s WHERE TENANT_ID = '%s' LIMIT 1", targetTableName, sourceAndTargetTenant));
+        deleteTargetStmt.execute();
+        conn.commit();
+        assertEquals(9, countUsers(targetTableName, sourceAndTargetTenant));
+        verify(String.format("TENANT_ID = '%s'", sourceAndTargetTenant), 9, 1, 1, 0, 0);
+
+    }
+
+    private void verify(String sqlConditions, int good, int bad, int onlyInSource, int onlyInTarget,
+            int contentDifferent) throws Exception {
+        VerifyReplicationTool.Builder builder =
+                VerifyReplicationTool.newBuilder(getUtility().getConfiguration());
+        builder.tableName(sourceTableName);
+        builder.targetTableName(targetTableName);
+        builder.sqlConditions(sqlConditions);
+        builder.timestamp(EnvironmentEdgeManager.currentTimeMillis());
+        VerifyReplicationTool tool = builder.build();
+        Job job = tool.createSubmittableJob();
+        // use the local runner and cleanup previous output
+        job.getConfiguration().set("mapreduce.framework.name", "local");
+        cleanupPreviousJobOutput(job);
+        Assert.assertTrue("Job should have completed", job.waitForCompletion(true));
+        Counters counters = job.getCounters();
+        LOG.info(counters.toString());
+        assertEquals(good,
+                counters.findCounter(VerifyReplicationTool.Verifier.Counter.GOODROWS).getValue());
+        assertEquals(bad,
+                counters.findCounter(VerifyReplicationTool.Verifier.Counter.BADROWS).getValue());
+        assertEquals(onlyInSource, counters.findCounter(
+                VerifyReplicationTool.Verifier.Counter.ONLY_IN_SOURCE_TABLE_ROWS).getValue());
+        assertEquals(onlyInTarget, counters.findCounter(
+                VerifyReplicationTool.Verifier.Counter.ONLY_IN_TARGET_TABLE_ROWS).getValue());
+        assertEquals(contentDifferent, counters.findCounter(
+                VerifyReplicationTool.Verifier.Counter.CONTENT_DIFFERENT_ROWS).getValue());
+    }
+
+    private void upsertData() throws SQLException {
+        Connection conn = DriverManager.getConnection(getUrl());
+        PreparedStatement sourceStmt =
+                conn.prepareStatement(String.format(UPSERT_USER, sourceTableName));
+        PreparedStatement targetStmt =
+                conn.prepareStatement(String.format(UPSERT_USER, targetTableName));
+        // create 2 tenants with 10 users each in source table
+        for (int t = 0; t < 2; t++) {
```

Comment:
nit: you can simplify this by creating 2 separate loops, just like the one you did on Line 183-185. That will be easier to understand.